### PR TITLE
Added unit test for post excerpt block render function

### PR DESCRIPTION
**Pull request by akasunil**
## What?
 
Added unit test for Post excerpt core block render function.

### CODE REVIEW

**Comment by akasunil** [comment]
@draganescu Can you please review this PR ?

**Comment by akasunil** [comment]
waiting for PR review

File: phpunit/blocks/render-block-post-excerpt-test.php
Comment (anton-vlasenko) [inline phpunit/blocks/render-block-post-excerpt-test.php]:
```suggestion
class Tests_Blocks_GutenbergRenderBlockCorePostExcerpt extends WP_UnitTestCase {
```

Comment (anton-vlasenko) [reply phpunit/blocks/render-block-post-excerpt-test.php]:
```
Test class names should reflect the filepath, with underscores replacing directory separators and TitleCase replacing camelCase. Thus, the class Tests_Comment_GetCommentClass(), which contains tests for the get_comment_class() function, is located in tests/comment/getCommentClass.php.
```
Please see https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#test-classes

File: phpunit/blocks/render-block-post-excerpt-test.php
Comment (anton-vlasenko) [inline phpunit/blocks/render-block-post-excerpt-test.php]:
```suggestion
 * @subpackage Blocks
 * @covers ::gutenberg_render_block_core_post_excerpt
```

File: phpunit/blocks/render-block-post-excerpt-test.php
Comment (anton-vlasenko) [inline phpunit/blocks/render-block-post-excerpt-test.php]:
```suggestion
		$rendered = gutenberg_render_block_core_post_excerpt( self::$attributes, '', $block );
```
Why does this function use capital letters in its name?
Is it a typo?

File: phpunit/blocks/render-block-post-excerpt-test.php
Comment (anton-vlasenko) [inline phpunit/blocks/render-block-post-excerpt-test.php]: Same.

File: phpunit/blocks/render-block-post-excerpt-test.php
Comment (anton-vlasenko) [inline phpunit/blocks/render-block-post-excerpt-test.php]: Same

File: phpunit/blocks/render-block-post-excerpt-test.php
Comment (anton-vlasenko) [inline phpunit/blocks/render-block-post-excerpt-test.php]: When a test method contains more than 1 assertion, a failure message should be added to the assertion method.
Please see https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#using-assertions

File: phpunit/blocks/render-block-post-excerpt-test.php
Comment (anton-vlasenko) [inline phpunit/blocks/render-block-post-excerpt-test.php]: Please add failure messages.

**Review by anton-vlasenko — COMMENTED** [review]
I've reviewed this PR. You can find my notes below.
Thanks!

**Comment by akasunil** [comment]
@anton-vlasenko Feedback addressed. Can you please review it again.

**Review by anton-vlasenko — COMMENTED** [review]
I've rebased this PR to trunk (locally) and I'm getting the following errors:
```
1) Tests_Blocks_GutenbergRenderBlockCorePostExcerpt::test_gutenberg_render_block_core_post_excerpt_empty
Undefined index: excerptLength

gutenberg/build/block-library/blocks/post-excerpt.php:27
gutenberg/phpunit/blocks/render-block-post-excerpt-test.php:99
phpvfscomposer://gutenberg/vendor/phpunit/phpunit/phpunit:106

2) Tests_Blocks_GutenbergRenderBlockCorePostExcerpt::test_gutenberg_render_block_core_post_excerpt
Undefined index: excerptLength

gutenberg/build/block-library/blocks/post-excerpt.php:27
gutenberg/phpunit/blocks/render-block-post-excerpt-test.php:121
phpvfscomposer://gutenberg/vendor/phpunit/phpunit/phpunit:106
```
Could you please have a look? Do you get the same error messages, @sunil25393 ?

File: phpunit/blocks/render-block-post-excerpt-test.php
Comment (anton-vlasenko) [inline phpunit/blocks/render-block-post-excerpt-test.php]:
```suggestion
/**
 * Tests the gutenberg_render_block_core_post_excerpt() function.
 *
 * @package WordPress
 * @subpackage Blocks
 *
 * @group blocks
 *
 * @covers ::gutenberg_render_block_core_post_excerpt
 */
```

File: phpunit/blocks/render-block-post-excerpt-test.php
Comment (anton-vlasenko) [inline phpunit/blocks/render-block-post-excerpt-test.php]:
```suggestion
	/**
	 * Test gutenberg_render_block_core_post_excerpt() method
	 * with empty data.
	 */
```

File: phpunit/blocks/render-block-post-excerpt-test.php
Comment (anton-vlasenko) [inline phpunit/blocks/render-block-post-excerpt-test.php]:
```suggestion
	/**
	 * Test gutenberg_render_block_core_post_excerpt() method.
	 */
```

**Review by anton-vlasenko — COMMENTED** [review]
https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#test-classes
> Test files should be named using camelCase.

**Comment by akasunil** [comment]
I have updated phpunit config temporary until we change all filenames to camel case as per wordpress core standard. And Addressed all feedback provided by reviewer.

File: phpunit/blocks/renderBlockCorePostExcerpt.php
Comment (anton-vlasenko) [inline phpunit/blocks/renderBlockCorePostExcerpt.php:21]:
```suggestion
	 * Array of attributes.
```

File: phpunit/blocks/renderBlockCorePostExcerpt.php
Comment (anton-vlasenko) [inline phpunit/blocks/renderBlockCorePostExcerpt.php]:
```suggestion
			'Failed to assert that $rendered contains the expected string.'
```

File: phpunit/blocks/renderBlockCorePostExcerpt.php
Comment (anton-vlasenko) [inline phpunit/blocks/renderBlockCorePostExcerpt.php]:
```suggestion
		$this->assertNotEmpty( $rendered, 'Failed to assert that $rendered is not empty.' );
```

File: phpunit/blocks/renderBlockCorePostExcerpt.php
Comment (anton-vlasenko) [inline phpunit/blocks/renderBlockCorePostExcerpt.php]:
```suggestion
			'Failed to assert that $rendered contains a closing html paragraph tag.'
```

File: phpunit/blocks/renderBlockCorePostExcerpt.php
Comment (anton-vlasenko) [inline phpunit/blocks/renderBlockCorePostExcerpt.php]:
```suggestion
			'Failed to assert that $rendered contains the "wp-block-post-excerpt__excerpt" string.'
```

File: phpunit/blocks/renderBlockCorePostExcerpt.php
Comment (anton-vlasenko) [inline phpunit/blocks/renderBlockCorePostExcerpt.php]:
```suggestion
			'Failed to assert that $rendered does not contain the has-text-align class.'
```

File: phpunit/blocks/renderBlockCorePostExcerpt.php
Comment (anton-vlasenko) [inline phpunit/blocks/renderBlockCorePostExcerpt.php]:
```suggestion
			'Failed to assert that $rendered contains the "has-text-align-left" class.'
```

File: phpunit/blocks/renderBlockCorePostExcerpt.php
Comment (anton-vlasenko) [inline phpunit/blocks/renderBlockCorePostExcerpt.php]:
```suggestion
			'Failed to assert that $rendered contains the expected string.'
```

File: phpunit/blocks/renderBlockCorePostExcerpt.php
Comment (anton-vlasenko) [inline phpunit/blocks/renderBlockCorePostExcerpt.php]:
```suggestion
			'Failed to assert that $rendered contains the expected string.'
```

File: phpunit/blocks/renderBlockCorePostExcerpt.php
Comment (anton-vlasenko) [inline phpunit/blocks/renderBlockCorePostExcerpt.php]:
```suggestion
			'Failed to assert that $rendered contain the expected post URL.'
```

File: phpunit/blocks/renderBlockCorePostExcerpt.php
Comment (anton-vlasenko) [inline phpunit/blocks/renderBlockCorePostExcerpt.php]: It's better to use `assertSame()` since `$rendered === '0'` will also pass the test, but obviously `$rendered` must be just an empty string.
```suggestion
		$this->assertSame( '', $rendered, 'Failed to assert that $rendered is an empty string.' );
```

File: phpunit/blocks/renderBlockCorePostExcerpt.php
Comment (anton-vlasenko) [inline phpunit/blocks/renderBlockCorePostExcerpt.php]: This seems like a duplicate assertion.

```suggestion
		$this->assertStringContainsString(
			'</p>',
			$rendered,
			'Failed to assert that $rendered contain html paragraph tag.'
		);
```

File: phpunit/blocks/renderBlockCorePostExcerpt.php
Comment (anton-vlasenko) [inline phpunit/blocks/renderBlockCorePostExcerpt.php]: I suggest using the word `should` in the test method names, as shown in the examples here: https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#test-methods

```suggestion
	public function test_should_render_empty_string_when_excerpt_is_empty() {
```
This is not a hard requirement, but rather a naming convention followed by some committers.
P.S. I think the function name can be omitted since this test class only tests one function.

File: phpunit/blocks/renderBlockCorePostExcerpt.php
Comment (anton-vlasenko) [inline phpunit/blocks/renderBlockCorePostExcerpt.php]: Same as above. Also, just a suggestion.
```suggestion
	public function test_should_render_correct_exceprt() {
```

File: phpunit/blocks/renderBlockCorePostExcerpt.php
Comment (anton-vlasenko) [inline phpunit/blocks/renderBlockCorePostExcerpt.php]: I've just applied some formatting.

```suggestion
/**
 * Tests for the gutenberg_render_block_core_post_excerpt() function.
 *
 * @package WordPress
 * @subpackage Blocks
 *
 * @covers ::gutenberg_render_block_core_post_excerpt
 * @group blocks
 */
```

**Review by anton-vlasenko — COMMENTED** [review]
These comments are mostly minor (sorry for being so meticulous). Thank you for considering my review.

**Comment by akasunil** [comment]
@anton-vlasenko I have addressed your feedbacks.

**Review by anton-vlasenko — APPROVED** [review]
Since each test method initializes the `$block` variable, I think it could be moved to the `set_up()` method.
However, I don't think this should be a blocker.
LGTM.

**Comment by akasunil** [comment]
Can we merge this PR ?

**Comment by akasunil** [comment]
@draganescu if you have space, can you review this PR too https://github.com/WordPress/gutenberg/pull/42435

**Comment by anton-vlasenko** [comment]
> Can we merge this PR ?

I apologize for assuming you could merge it yourself.
I'm glad to see it has been merged now.

**Comment by akasunil** [comment]
> I apologize for assuming you could merge it yourself.

No worries

**Pull request merged**
